content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+
+```{r}
+
+##### THESE FIRST TWO SEARCHES STILL NEED WORK #####
+
+# search by keyword
+precip_query <- list(q = 'precipitation')
+precip_data <- query_sb(query_list = precip_query)
+length(precip_data) # 50 entries, so there is likely more than 50 results
+head(precip_data, 2)
+
+# search by keyword + category
+precip_maps_query <- list(q = 'precipitation', browseType = "Static Map Image", sort='title')
+precip_maps_data <- query_sb(query_list = precip_query)
+head(precip_maps_data, 2)
+
+##### -------------------------------------- #####
+
+# search by 2 keywords
+hazard2_query <- list(q = 'flood', q = 'earthquake')
+hazard2_data <- query_sb(query_list = hazard2_query)
+length(hazard2_data)
+head(hazard2_data, 2)
+
+# search by 3 keywords
+hazard3_query <- list(q = 'flood', q = 'earthquake', q='drought')
+hazard3_data <- query_sb(query_list = hazard3_query)
+length(hazard3_data)
+head(hazard3_data, 2)
+```
+
+### Using `query_sb_text`
+
+`query_sb_text` returns a list of `sbitems` that match the title or description fields. Use it to search authors, station names, rivers, states, etc.
+
+```{r}
+# search using a contributors name
+contrib_results <- query_sb_text("Robert Hirsch")
+head(contrib_results, 2)
+
+# search using place of interest
+park_results <- query_sb_text("Yellowstone")
+head(park_results, 2)
+
+# search using a site location
+loc_results <- query_sb_text("Embudo")
+length(loc_results)
+head(loc_results, 2)
+```
+
+### Using `query_sb_doi`
+
+Use a Digital Object Identifier (DOI) to query ScienceBase. This should return only one list item, unless there is more than one ScienceBase item referencing this very unique identifier.
+
+```{r}
+# USGS Microplastics study 
+query_sb_doi('10.5066/F7ZC80ZP')
+
+####### I've tried A TON of DOIs I found through the web interface and just
+####### keep getting empty lists returned.
+query_sb_doi('10.1016/j.coldregions.2007.05.009')
+```
+
+### Using `query_sb_spatial`
+
+`query_sb_spatial` accepts 3 different methods for specifying a spatial area in which to look for data. You can specify a bounding box `bbox` as an `sp` spatial data object *[[[[NEED MORE TEXT HERE]]]]*.  Alternatively, you can supply a vector of latitudes and a vector of longitudes using `lat` and `long` arguments. The function will automatically use the minimum and maximum from those vectors to construct a boundary box. The last way to represent a spatial region to query ScienceBase is using a POLYGON Well-known text (WKT) object as a text string. The format is `"POLYGON(([LONG1 LAT1], [LONG2 LAT2], [LONG3 LAT3]))"`, where `LONG#` and `LAT#` are longitude and latitude pairs as decimals. See the [Open Geospatial Consortium WKT standard](http://www.opengeospatial.org/standards/wkt-crs) for more information. 
```

Comment:
the `MORE TEXT` tag was because I thought I needed to explain what an sp object is. Or at least link out to something that describes it